lib/entry.js

```diff
@@ -170,75 +157,34 @@ Ep.runModuleSetters = function (module) {
     }
   }
 
-  // Every three elements of this list form a (setter, value, name) triple
-  // that needs to be invoked.
-  var settersToCall = [];
-
-  // Lazily-initialized objects mapping parent module identifiers to
-  // relevant parent module objects and snapshots of their exports.
+  // Lazily-initialized object mapping parent module identifiers to parent
+  // module objects whose setters we might need to run.
   var relevantParents;
-  var parentSnapshots;
 
   // Take snapshots of setter.parent.exports for any setters that we are
   // planning to call, so that we can later determine if calling the
   // setters modified any of those exports objects.
   forEachSetter(function (setter, value, name) {
-    var parent = setter.parent;
-    parentSnapshots = parentSnapshots || Object.create(null);
-    if (! hasOwn.call(parentSnapshots, parent.id)) {
-      relevantParents = relevantParents || Object.create(null);
-      relevantParents[parent.id] = parent;
-      if (utils.isPlainObject(parent.exports)) {
-        // If parent.exports is an object, make a shallow clone of it so
-        // that we can see if it changes as a result of calling setters.
-        parentSnapshots[parent.id] = Object.assign({}, parent.exports);
-      } else {
-        // If parent.exports is not an object, the "snapshot" is just the
-        // value of parent.exports.
-        parentSnapshots[parent.id] = parent.exports;
-      }
-    }
-
-    // Push three elements at a time to avoid creating wrapper arrays for
-    // each (setter, value, name) triple. Note the i += 3 below.
-    settersToCall.push(setter, value, name);
-  });
-
-  // Now call all the setters that we decided we need to call.
-  for (var i = 0; i < settersToCall.length; i += 3) {
-    var setter = settersToCall[i];
-    var value = settersToCall[i + 1];
-    var name = settersToCall[i + 2];
+    relevantParents = relevantParents || Object.create(null);
+    relevantParents[setter.parent.id] = setter.parent;
     setter.call(module, setter.last[name] = value, name);
-  }
+  });
 
   ++entry.runCount;
 
   if (! relevantParents) {
-    // If we never called takeSnapshot, then we can avoid checking
-    // relevantParents and parentSnapshots below.
     return;
   }
 
   // If any of the setters updated the module.exports of a parent module,
   // or updated local variables that are exported by that parent module,
   // then we must re-run any setters registered by that parent module.
   Object.keys(relevantParents).forEach(function (id) {
-    var parent = relevantParents[id];
```

Comment:
I'd caution use of  `Array#forEach` and other ES5/6 functional array helpers in code that you'd like to be snappy. One off calls are OK but when you stack them they can bog down execution *(engines haven't done a great job of optimizing them even 12 yrs later)*. With `let`, `const`, and `for-of` it allows many of the readability and logic-guard wins of `Array#forEach` and friends with less cumulative overhead.